src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java

```diff
@@ -179,12 +188,34 @@ private void ingestActionsPerUser() {
 
     private void ingestBillPayUsers() {
         if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY)) {
-            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport()
-                .stream()
-                .map(LegalEntityWithUsers::getUserExternalIds)
-                .flatMap(List::stream)
-                .collect(Collectors.toList())
-                .forEach(this.billpayConfigurator::ingestBillPayUser);
+            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport().forEach((le) -> {
+
+                List<Account> accounts = new ArrayList<>();
+                String externalUserId = le.getUserExternalIds().get(0);
+                loginRestClient.login(externalUserId, externalUserId);
+                String externalLegalEntityId = this.userPresentationRestClient.
+                    retrieveLegalEntityByExternalUserId(externalUserId)
+                    .getExternalId();
+                List<ArrangementItem> arrangementItems = accountsIntegrationRestClient
+                    .getArrangements(externalLegalEntityId);
+
+                arrangementItems.forEach((arrangement) -> {
+                    accounts.add(mapBillPayAccount(arrangement));
+                });
+                le.getUserExternalIds().forEach((user) -> {
+                    billpayConfigurator.ingestBillPayUser(user);
+                    if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY_ACCOUNTS)) {
+                        billpayConfigurator.ingestBillPayAccounts(user, accounts);
```

Comment:
for a company or organizatino i can imagine that accounts are being shared between employees, can you take that scenario into account here?

Reply:
I'm not entirely sure what you mean here. If two users have access to the same accounts, then the same account details will be ingested for each user into Bill Pay, as expected.

Reply:
cool, then it's okay. i got this idea after reading `ingestAccounts` in the configurator: 
```    .....
        UserByIdPutRequestBody request = new UserByIdPutRequestBody()
            .withSubscriber(createSubscriber(subscriberId, accounts));
        Response response = billPayMockBankRestClient.ingestAccounts(request);
```